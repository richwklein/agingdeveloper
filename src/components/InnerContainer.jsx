import React from "react";
import {Container} from "@mui/material";
import PropTypes from "prop-types";
import {ChildrenType} from "../types";

// TODO remove spread
const InnerContainer = ({children, isMain=false}) => {
  const component = isMain ? "main" : "div";

  return (
    <Container maxWidth="lg" component={component}>
      {children}
    </Container>
  );
};

InnerContainer.propTypes = {
  children: ChildrenType,
  isMain: PropTypes.bool,
};

export default InnerContainer;
