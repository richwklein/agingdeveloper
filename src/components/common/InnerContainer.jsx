import React from "react";
import {Container} from "@mui/material";
import PropTypes from "prop-types";

const InnerContainer = ({props, children}) => {
  return (
    <Container maxWidth="lg" {...props}>
      {children}
    </Container>
  );
};

InnerContainer.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
  ]),
  props: PropTypes.any,
};

export default InnerContainer;
