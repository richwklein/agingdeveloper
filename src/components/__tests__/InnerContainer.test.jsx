import React, {Fragment} from "react";
import {cleanup, render} from "@testing-library/react";
import InnerContainer from "../InnerContainer";

describe("InnerContainer", () => {
  afterEach(cleanup);

  const renderComponent = () => render(
      <InnerContainer>
        <Fragment>Test</Fragment>
      </InnerContainer>,
  );

  test("that it renders correctly.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("that it renders with main.", () => {
    const {container} = render(<InnerContainer isMain={true}>Test</InnerContainer>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("that it has a maxWidth of large.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toHaveClass("MuiContainer-maxWidthLg");
  });
});
