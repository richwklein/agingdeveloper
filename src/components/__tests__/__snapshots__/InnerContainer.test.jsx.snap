// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InnerContainer that it renders correctly. 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
>
  Test
</div>
`;

exports[`InnerContainer that it renders with main. 1`] = `
<main
  class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
>
  Test
</main>
`;
